repo.go

```diff
@@ -438,38 +443,44 @@ func (r *Repository) processItem(ctx itemContext) error {
 	} else {
 		// we already have this item in the DB
 
-		// if we don't have it on disk as a file or in the media list file for
-		// this collection already, add path to text file in this collection.
-		folderHas, err := r.localCollectionHasItemOnDisk(ctx.ac.account, ctx.coll, loadedItem)
-		if err != nil {
-			return fmt.Errorf("checking if local collection has item: %v", err)
-		}
 		_, dbHas := loadedItem.Collections[ctx.coll.CollectionID()]
-		if !folderHas && !dbHas {
-			// okay, the fact that this item belongs to this collection is
-			// new information. add it to the media list file and save it
-			// to the collection in the DB.
-			err := r.writeToMediaListFile(ctx.coll, loadedItem.FilePath)
-			if err != nil {
-				return fmt.Errorf("writing to media list file: %v", err)
+		corrupted := false
+
+		if !dbHas || !ctx.skipRepoChecks {
+			// if we don't have it on disk as a file or in the media list file for
+			// this collection already, add path to text file in this collection.
+			if folderHas, err := r.localCollectionHasItemOnDisk(ctx.ac.account, ctx.coll, loadedItem); err != nil {
+				return fmt.Errorf("checking if local collection has item: %v", err)
+			} else if !folderHas {
+				if err := r.writeToMediaListFile(ctx.coll, loadedItem.FilePath); err != nil {
```

Comment:
We always write the file path to media list first and only after that - to the database. So, if `dbHas` is `true`, we can be sure that it's already in the media list.

`stat()`ing a few thousand files and reading media lists for each deduplicated file is expensive enough to optimize it, I believe.